File: Assignment18/Q2.java
///////////////////////////////////////////////////////////////////
//
//      File Name : Q2.java
//      Project   : Logic Building Assignment 18
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : EvenNumberLogic
//      Method Name: printEvenNumbers
//      Input      : Integer
//      Output     : Void
//      Description: Print all even numbers up to N
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class EvenNumberLogic
{
    void printEvenNumbers(int n)
    {
        int i = 0;      // Loop counter

        // Print all even numbers from 1 to n
        for(i = 1; i <= n; i++)
        {
            if(i % 2 == 0)
            {
                System.out.print(i + " ");
            }
        }
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q2
{
    public static void main(String args[])
    {
        // Create object of EvenNumberLogic class
        EvenNumberLogic obj = new EvenNumberLogic();

        // Function call
        obj.printEvenNumbers(20);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 20
//
//      Output :
//      2 4 6 8 10 12 14 16 18 20
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(n)
//
//      Space Complexity : O(1)
//
///////////////////////////////////////////////////////////////////

File: Assignment23/Q2.c
///////////////////////////////////////////////////////////////////
//
//      File Name : Q2.c
//      Project   : Logic Building Pattern Assignment
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Including Header File
//
///////////////////////////////////////////////////////////////////

#include<stdio.h>

///////////////////////////////////////////////////////////////////
//
//      Function Name : Pattern
//      Input         : Integer, Integer
//      Output        : Void
//      Description   : Display numbers from 1 to
//                      column count for every row
//      Date          : 08/06/2026
//      Author        : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

void Pattern(int iRow, int iCol)
{
    int i = 0;      // Row counter
    int j = 0;      // Column counter

    // Traverse each row
    for(i = 1; i <= iRow; i++)
    {
        // Print numbers from 1 to iCol
        for(j = 1; j <= iCol; j++)
        {
            printf("%d ", j);
        }

        // Move cursor to next line
        printf("\n");
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Function
//
///////////////////////////////////////////////////////////////////

int main()
{
    int iValue1 = 0;      // Store number of rows
    int iValue2 = 0;      // Store number of columns

    // Accept rows and columns from user
    printf("Enter number of rows and columns : ");
    scanf("%d %d", &iValue1, &iValue2);

    // Function call
    Pattern(iValue1, iValue2);

    return 0;
}

///////////////////////////////////////////////////////////////////
//
//      Input  :
//      Rows = 4
//      Columns = 3
//
//      Output :
//      1 2 3
//      1 2 3
//      1 2 3
//      1 2 3
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(iRow * iCol)
//
//      Space Complexity : O(1)
//
///////////////////////////////////////////////////////////////////

